x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -135,11 +146,37 @@ export class RuleDataClient implements IRuleDataClient {
           },
         });
       } catch (err) {
-        // something might have created the index already, that sounds OK
-        if (err?.meta?.body?.error?.type !== 'resource_already_exists_exception') {
+        // If the index already exists and it's the write index for the alias,
+        // something else created it so suppress the error. If it's not the write
+        // index, that's bad, throw an error.
+        if (err?.meta?.body?.error?.type === 'resource_already_exists_exception') {
+          const { body: existingIndices } = await clusterClient.indices.get({
+            index: concreteIndexName,
+          });
+          if (!existingIndices[concreteIndexName]?.aliases?.[alias]?.is_write_index) {
+            throw Error(
```

Comment:
as of ES5 they're supposed to be equivalent, right? https://es5.github.io/#x15.11.1

Reply:
Yes, but for a custom exception this might not work.

Example:

```js
class CustomError extends Error {
  constructor(value) {
    super('Hello this is my message');
    this.value = value;
  }
}

let e = CustomError(42);
console.log(e.value); // ?
```

Actually, I just checked it in Chrome, and calling `CustomError(42)` without `new` just throws `TypeError: Class constructor CustomError cannot be invoked without 'new'`, seems like V8 has a special handling for `class` constructors.

However, this won't work the same way for any constructor defined as a function. In general, something like that would be needed:

```js
function User(name) {
  if (!new.target) { // if you run me without new
    return new User(name); // ...I will add new for you
  }

  this.name = name;
}

let john = User("John"); // redirects call to new User
alert(john.name); // John
```

See https://javascript.info/constructor-new#constructor-mode-test-new-target

Maybe I was overthinking when trying to explain it, but for me it's quite simple - let's use `new` everywhere and in all cases when calling constructors for the sake of safety and consistency. It's like `===`.

Sorry, I'm 🚲 🏠 🎨 'ing